spec/unit/shoulda/matchers/active_record/have_rich_text_matcher_spec.rb

```diff
@@ -0,0 +1,137 @@
+require 'unit_spec_helper'
+
+describe Shoulda::Matchers::ActiveRecord::HaveRichText, type: :model do
+  def self.rich_text_is_defined?
+    defined?(ActionText::RichText)
+  end
+
+  if rich_text_is_defined?
+    it 'returns the message including the name of the provided attribute' do
+      matcher = have_rich_text(:content)
+      expect(matcher.description).
+        to eq('have :content as a ActionText::RichText')
+    end
+
+    context 'a has_rich_text association' do
+      it 'matches when the subject configures has_rich_text for a custom ' \
+        'attribute' do
+        create_table_posts
+    
+        valid_record = new_post(has_rich_text: true)
+    
+        expect(valid_record).to have_rich_text(:content)
+      end
+    
+      it 'does not matches when lacking has_rich_text ' \
+          'attribute' do
+        create_table_posts
+    
+        invalid_record = new_post(has_rich_text: false)
+        expected_message = 'Expected Post to have :invalid_attribute as a ' \
+          'ActionText::RichText but :invalid_attribute does not exist'
+    
+        aggregate_failures do
+          expect(invalid_record).not_to have_rich_text(:invalid_attribute)
+          expect { expect(invalid_record).to have_rich_text(:invalid_attribute) }.
+            to fail_with_message(expected_message)
+        end
+      end
+    
+      it 'does not matches when the provided attribute it\'s not a ' \
+          'RichText class' do
+        create_table_posts(has_content_attribute: true)
+    
+        invalid_model = define_model_class(:Post)
+        invalid_record = invalid_model.new(content: "Content")
+    
+        expected_message = 'Expected Post to have :content as a ' \
+          'ActionText::RichText but :content is a String'
+    
+        aggregate_failures do
+          expect(invalid_record).not_to have_rich_text(:content)
+          expect { expect(invalid_record).to have_rich_text(:content) }.
+            to fail_with_message(expected_message)
+        end
+      end
+    
+      it 'fails with the appropriate message when negated' do
+        expect { expect(new_post(has_rich_text: true)).not_to have_rich_text(:content) }.
+          to fail_with_message('Did not expect Post to have a rich text :content')
+      end
+    end
+  elsif active_record_version >= 4 && active_record_version < 5
+    it 'raise a undefined method error when action text is not defined' do
+      create_table_posts
+  
+      invalid_record = -> { new_post(has_rich_text: true) }
+
+      message = /undefined method `has_rich_text' for Post/
+        
+      expect(&invalid_record).to raise_error(
+        NoMethodError,
+        message
+      )
+    end
+
+    it 'does not matches when the provided attribute it\'s not a ' \
```

Comment:
What if you used double quotes here? I feel like backslashing the apostrophe makes it less readable.

Reply:
True. Done.